### DISCUSSION TITLE
Custom env variable for custom domains

### DISCUSSION THREAD

**Discussion by akinwol** [vercel/community#26] [Help] [answered]
Hi everyone, newbie here so bare with me. 

I am working with a couple customers as part of a beta for an idea and need help. I am looking to build a Saas based solution/platform. Launched a beta for 1 customer but it is not scalable AT ALL.

**What I need**

1. Custom/sub domain for each customer (which I found a solution for [here](https://github.com/vercel/vercel/discussions/4840))
2. Custom env variables for each customer (unique env variables per custom domain). 
   - Each customer in some cases will need their own env variables for third party integrations. For e.g. something like a custom Shopify store each customer would have their own keys/secrets to connect to their store. Same goes for any available integrations we might have.
3. Is there a way to add new env like variables dynamically post build? For eg. if I have a production build running but a customer wants to add a new API Key for an integration they want to use. Does Vercel have any integrations to dynamically store secrets for each custom domain? Or is there a recommended solution to solve for something like this?
4. Configs per customer/custom domain - What is the best way with Vercel to load up custom configs such as themes, available feature, meta tags etc for each customer. Can this be done dynamically or can it only be done for each build. 

I have considered creating a separate branch for each customer but I would think this could get messy as the app grows. If I have 1000+ customers, I don't know if I would want to maintain 1000+ branches. But if this is the best way the so be it.

Any advice or help would be greatly appreciated

**Comment by steven-tey** [comment, marked as answer]
You could try storing the ENV vars in a secure DB and pull them in on the server-side (e.g. getStaticProps) at build time and render the content accordingly. That way you get to leverage ISR since you can now perform CRUD on your ENV vars post build time and have that regenerate incrementally.

**Reply by akinwol** [reply]
@steven-tey  how can you update ENV var post build, do you have code sample for that? New to this so that would be helpful.  In my mind if I was just using getStatisProps to pull env var and I need to use it I would need to make another call to pull the data I need with the env var wherever its needed.

**Reply by akinwol** [reply]
@steven-tey  also can you modify public env variables? Basically I'm working on a project with a few small businesses. Building a custom e-commerce storefront. I just want to build 1 they can all use and configure. But in order to connect to say their e-commerce backend I need to have separate env variables on both the client and server side. Seems like public env variables can only be changed at build time ?
